src/nl/hannahsten/texifyidea/index/file/LatexDtxDefinitionDataIndexer.kt

```diff
@@ -0,0 +1,360 @@
+package nl.hannahsten.texifyidea.index.file
+
+import com.intellij.util.indexing.DataIndexer
+import com.intellij.util.indexing.DefaultFileTypeSpecificInputFilter
+import com.intellij.util.indexing.FileBasedIndex
+import com.intellij.util.indexing.FileBasedIndexExtension
+import com.intellij.util.indexing.FileContent
+import com.intellij.util.indexing.ID
+import com.intellij.util.io.DataExternalizer
+import com.intellij.util.io.EnumeratorStringDescriptor
+import com.intellij.util.io.IOUtil
+import com.intellij.util.io.KeyDescriptor
+import nl.hannahsten.texifyidea.file.LatexSourceFileType
+import nl.hannahsten.texifyidea.index.LatexFileBasedIndexKeys
+import nl.hannahsten.texifyidea.index.file.LatexSimpleDefinition.Type.COMMAND
+import nl.hannahsten.texifyidea.index.file.LatexSimpleDefinition.Type.ENVIRONMENT
+import nl.hannahsten.texifyidea.lang.LArgument
+import nl.hannahsten.texifyidea.lang.LArgumentType
+import nl.hannahsten.texifyidea.lang.LatexLib
+import nl.hannahsten.texifyidea.util.Log
+import java.io.DataInput
+import java.io.DataOutput
+
+/**
+ * A simple definition of a LaTeX command or environment.
+ */
+data class LatexSimpleDefinition(
+    /**
+     * Without the leading backslash.
+     */
+    var name: String,
+    var type: Type,
+    var arguments: List<LArgument> = emptyList(),
+    var description: String = ""
+) {
+
+    val isEnv: Boolean
+        get() = type == ENVIRONMENT
+
+    override fun toString(): String {
+        val displayName = if(isEnv) name else "\\$name"
+        return "Def('$displayName', ${arguments.joinToString("")}, desc='${description.take(20)}')"
+    }
+
+    enum class Type {
+        COMMAND,
+        ENVIRONMENT
+    }
+}
+
+/**
+ *
+ */
+object LatexDtxDefinitionDataIndexer : DataIndexer<String, List<LatexSimpleDefinition>, FileContent> {
```

Comment:
The documentation does not work for me in the completion popup, like in the screenshot in https://hannah-sten.github.io/TeXiFy-IDEA/code-completion.html#autocomplete-installed-commands
It does work when requesting documentation for a command after it is put in the document